src/index.ts

```diff
@@ -18,47 +21,102 @@ const seed = randomBytes(32)
 // Global services that are set up once and then available in all integration tests
 declare global {
     const ceramic: CeramicApi
+    const ceramic2: CeramicApi
+}
+
+const buildCeramic = async (configObj, ipfs: IpfsApi): Promise<CeramicApi> => {
+    let ceramic
+    if (configObj.mode == "client") {
+        console.log(`Creating ceramic via http client, connected to ${configObj.apiURL}`)
+        ceramic = new CeramicClient(configObj.apiURL, { docSyncEnabled: true, docSyncInterval: 500 })
+    } else if (configObj.mode == "node") {
+        console.log("Creating ceramic local node")
+        const ceramicConfig: CeramicConfig = {
+            networkName: configObj.network,
+            pubsubTopic: configObj.pubsubTopic,
+            ethereumRpcUrl: configObj.ethereumRpc,
+            anchorServiceUrl: configObj.anchorServiceAPI,
+        }
+        ceramic = await Ceramic.create(ipfs, ceramicConfig)
+    } else if (configObj.mode == "none") {
+        return null
+    } else {
+        throw new Error(`Ceramic mode "${configObj.mode}" not supported`)
+    }
+
+    const didProvider = new Ed25519Provider(seed)
+    await ceramic.setDIDProvider(didProvider)
+
+    return ceramic
+}
+
+const buildIpfs = async (configObj): Promise<IpfsApi> => {
+    let ipfs: IpfsApi
+    multiformats.multicodec.add(dagJose)
+    const format = legacy(multiformats, dagJose.name)
+    if (configObj.mode == "client") {
+        console.log(`Creating IPFS via http client, connected to ${configObj.apiURL}`)
+        ipfs = ipfsClient({url: configObj.apiURL, ipld: {formats: [format]}})
+    } else if (configObj.mode == "node") {
+        console.log(`Creating IPFS local node`)
+        const repoPath = (await tmp.dir()).path
+        ipfs = await IPFS.create({
+            repo: repoPath,
+            ipld: {
+                formats: [format]
+            },
+            libp2p: {
+                config: {
+                    dht: {
+                        enabled: true,
+                        clientMode: !configObj.dhtServerMode,
+                        randomWalk: false,
+                    },
+                },
+            },
+            config: {
+                Routing: {
+                    Type: configObj.dhtServerMode ? 'dhtserver' : 'dhtclient',
+                },
+            }
+        })
+    } else if (configObj.mode == "none") {
+        return null
+    } else {
+        throw new Error(`IPFS mode "${configObj.mode}" not supported`)
+    }
+
+    return ipfs
```

Comment:
For a different pr but I think we should import the ipfs config from js-ipfs-ceramic. Currently can't because it's not exported there but will be nice to include that here too as a default.

Reply:
Yep, agreed 100%.